### PR TITLE
wrap env prefix with double quotes

### PR DESCRIPTION
**Pull request by kejxu**
adding double quotes will support environment variables that contain space; meantime, it does not affect environment variables that does not contain space.

test case (not ROS-specific):
this will fail, `value: command not found`
```bash
someenv=somevalue someenv2=some value python -c "import os;print os.environ['someenv'];print os.environ['someenv2']"
```

this would work:
```bash
someenv="somevalue" someenv2="some value" python -c "import os;print os.environ['someenv'];print os.environ['someenv2']"
```

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Thanks for the patch.

**Comment by dirk-thomas** [comment]
@kejxu With #1809 merged this needs to be rebased to resolve the conflicts.

**Comment by kejxu** [comment]
> @kejxu With #1809 merged this needs to be rebased to resolve the conflicts.

merged `master` into feature branch (instead of rebasing) for quick resolution =)

File: tools/roslaunch/src/roslaunch/node_args.py [hunk 1]
```diff
@@ -198,9 +198,9 @@ def get_node_args(node_name, roslaunch_files):
         return ["set %s=%s&&"%(k, v) for k, v in env.items()] + args
     else:
         # sys.platform.startswith('linux')
-        # join environment vars are bash prefix args
-        return ["%s=%s"%(k, v) for k, v in env.items()] + args
-    
+        # join environment vars are bash prefix args, wrap with double quotes for variables that contains space
+        return [r'%s="%s"'%(k, v) for k, v in env.items()] + args
```
Comment (dirk-thomas) [inline tools/roslaunch/src/roslaunch/node_args.py]: Why is the r-prefix used here?

Comment (kejxu) [reply tools/roslaunch/src/roslaunch/node_args.py]: remove unnecessary r-prefix

**Comment by dirk-thomas** [comment]
Thanks for the patch.

**Pull request merged**